src/test/java/com/uber/profiling/YamlConfigProviderTest.java

```diff
@@ -76,14 +83,20 @@ public void getConfig() throws IOException {
             Assert.assertEquals(2, config.size());
 
             Map<String, List<String>> rootConfig = config.get("");
-            Assert.assertEquals(2, rootConfig.size());
+            Assert.assertEquals(6, rootConfig.size());
             Assert.assertEquals(Arrays.asList("value1"), rootConfig.get("key1"));
-            Assert.assertEquals(Arrays.asList("value2a", "value2b"), rootConfig.get("key2"));
+            Assert.assertEquals(Arrays.asList("value2a"), rootConfig.get("key2[0]"));
+            Assert.assertEquals(Arrays.asList("value2b"), rootConfig.get("key2[1]"));
+            Assert.assertEquals(Arrays.asList("value3a"), rootConfig.get("key3.key3a"));
+            Assert.assertEquals(Arrays.asList("value3b"), rootConfig.get("key3.key3b[0]"));
```

Comment:
To keep backward compatible, we need: Assert.assertEquals(Arrays.asList("value3b", "value3c"), rootConfig.get("key3.key3b"));